server/etcdserver/server.go

```diff
@@ -1110,6 +1111,7 @@ func (s *EtcdServer) run() {
 	if s.lessor != nil {
 		expiredLeaseC = s.lessor.ExpiredLeasesC()
 	}
+	osutil.RegisterInterruptHandler(func() { s.triggerSnapshot(&ep) })
```

Comment:
Could you, please, add it to: 

https://github.com/etcd-io/etcd/blob/556447683112b6ccc00b27a1bd0d0a4edc361692/server/embed/etcd.go#L360

instead of registering a second InterruptHandler ?

I'm afraid that with the current approach we would have a 2 racy code-paths mutating the server state.

Reply:
Moved to etcdserver run defer function as it's the only place we have access to etcdProgress.